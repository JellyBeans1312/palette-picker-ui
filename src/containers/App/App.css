.main {
  display: flex;
  flex-wrap: wrap;
  height:100vh;
  width: 100%;
}

.form-container {
  display: flex;
  flex-direction: column;
  width: 50%;
}

.header {
  justify-content: center;
  height: 10%;
  text-align: center;
  width: 100%;
}

h1 {
  height: 30px;
  margin: 0;
}

.update-project-name {
  display: flex;
  justify-content: center;
}