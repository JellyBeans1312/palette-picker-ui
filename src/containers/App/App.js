import React, { Component } from 'react';
import CreateProjectForm from '../CreateProjectForm/CreateProjectForm';
import ColorContainer from '../ColorContainer/ColorContainer';
import { connect } from 'react-redux';
import { getAllProjects } from '../../util/apiCalls'
import { addAllProjects, saveColors } from '../../actions'
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      colors: []
    }
  }

  componentDidMount = async () => {
    const allProjects = await getAllProjects()
    this.props.addAllProjects(allProjects)
    this.hexCodeGenerator(this.state.colors)
  }

  hexCodeGenerator = (colors) => {
    if (colors.length > 4) {
      return colors;
    }

    let randomColor = { color: '#' + (Math.random() * 0xFFFFFF << 0).toString(16), isLocked: false};
    if (randomColor.color.length < 7) {
      randomColor = { color: '#' + (Math.random() * 0xFFFFFF << 0).toString(16), isLocked: false}
    }
    if(this.state.colors.length < 5) {
      this.setState({colors: [...this.state.colors, randomColor]});
    } 

    return this.hexCodeGenerator(this.state.colors);
  }

  generateNewColors = () => {
    let updatedColors = this.state.colors.map(color => {
      console.log(color.color)
      if(color.isLocked) {

      } else {
        console.log(color.color)
        return { color: '#' + (Math.random() * 0xFFFFFF << 0).toString(16), isLocked: false }
      }
    });
    console.log('UPDATED COLORS', updatedColors)
    this.setState({ colors: updatedColors })
  }

  render() {
    const { allProjects } = this.props;
    return (
      <main>
      {!allProjects && <CreateProjectForm />}
      <ColorContainer generateNewColors={this.generateNewColors} colors={this.state.colors}/>
      </main>
    )
  }
}

const mapStateToProps = state => ({
  allProjects: state.allProjects
});

const mapDispatchToProps = dispatch => ({
  addAllProjects: allProjects => dispatch(addAllProjects(allProjects)),
  saveColors: colors => dispatch(saveColors(colors))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
