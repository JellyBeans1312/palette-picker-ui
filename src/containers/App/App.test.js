import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App', () => {
  let wrapper;
  const mockColors = [{}]
  const mockProject = {}
  const mockAllProjects=[{}]

  beforeEach(() => {
    wrapper = shallow(
      <App
        colors={mockColors}
        project={mockProject}
        allProjects={mockAllProjects}
        addAllPalettes={jest.fn()}
        addAllProjects={jest.fn()}
        addPalette={jest.fn()}
        addProject={jest.fn()}
      />
    )
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})