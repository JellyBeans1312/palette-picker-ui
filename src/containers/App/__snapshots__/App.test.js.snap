// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<main
  className="main"
>
  <header
    className="header"
  >
    <h1>
      Welcome to Palette Picker
    </h1>
    <div
      className="update-project-name"
    >
      <p>
        asdf
      </p>
      <img
        alt="edit"
        className="edit-button"
        onClick={[Function]}
        src="editIcon.svg"
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <img
        alt="delete"
        className="delete-button"
        onClick={[Function]}
        src="xImage.svg"
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <div>
        <input
          onChange={[Function]}
          placeholder="asdf"
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
        >
          Update Project Name
        </button>
      </div>
    </div>
  </header>
  <Connect(ColorContainer)
    colors={
      Array [
        Object {
          "color": "#fff",
          "isLocked": false,
        },
        Object {
          "color": "#000",
          "isLocked": false,
        },
        Object {
          "color": "#ececec",
          "isLocked": false,
        },
        Object {
          "color": "#aeaeae",
          "isLocked": false,
        },
        Object {
          "color": "#d8d646",
          "isLocked": false,
        },
      ]
    }
    generateNewColors={[Function]}
    updatePalette={[Function]}
  />
  <Connect(ProjectContainer) />
  <div
    className="form-container"
  >
    <Connect(AddNewProjectForm) />
    <Connect(CreatePaletteForm) />
  </div>
</main>
`;
